### DISCUSSION TITLE
Capture points for Control Panel Exploration in-person session

### DISCUSSION THREAD

**Discussion by jacobwoffenden** [ministryofjustice/analytical-platform#1946] [General]
The purpose of this on-site collaboration session between Apps/Tools and Labs is to capture our desired user journey for Data Platform's Control Panel.

So we can fully utilise this time, we should create an agenda/points we want to cover.

[This Google Doc](https://docs.google.com/document/d/1F19rzGHz18yRnf0SF5224XXgsh8uBBjaRqbug-TVADY/edit?usp=sharing) has captured our thoughts so far.

Proposed Points:

- How a user signs into Control Panel
- How users/groups are reconciled across our services
- How Control Panel integrates with ingestion mechanism from Labs

Proposed Agenda:

- Mock up user flow for various journeys

**Comment by michaeljcollinsuk** [comment]
Will we have time to cover all of the flows from the google doc (session is only in the diary for 2.5 hours)? I would prioritise the "Your data", "Your Teams" and "Your Tools" flows

**Comment by ymao2** [comment]
I dump my ideas here, 
The user flows we may include: 
- Registering as a DP user
- Onboarding a DP user
- DP User registers a dataset (or data product?) (NOTES: should we use data product concept to user?), this flow should cover the integration with ingestion mechanism 
- DP User asks the access to a dataset (or data product?) and related approval flow

In order to close the lifecycle of each resource we manage, then ( those can be pushed down to another session)
- Off-boarding a user
- Removing a dataset(data product) flow
- Revoking/updating the access level for a user. 

The points the discussion should cover if possible 
- Steps in each user flow +/ major UI(?)

**For each step of flow, typical involve**
- What is entry condition and/or assumption?
- who can trigger this step?
- What input data is involved? where those data is from?
- who are the participants?  user(role) in Control panel or any tool/platform/service outside Control panel? how we integrate them?
- What are the outcomes from the step? where are they stored? 
  e.g. S3 bucket, AWS role , how to design the outcome to get user's setting ready for other requirements like launching tooling and viewing the metadata from Open metadata etc ---> the point `users/groups are reconciled across our services` by Jacob

**Comment by michaeljcollinsuk** [comment]
From @julialawrence via [slack](https://mojdt.slack.com/archives/C04M8224WCV/p1698130506008959):

We would like to understand how different types of users move through the proposed single data platform interface. Starting with data consumer similar to the one we service in AP, followed by data owner and additional personas.
Ones I particularly hope we can cover would be creating a data product (ingestion and catalogue registration,) searching for data (should this be done in the interface or routed to catalogue,) requesting access and granting access. I think getting on the same page regarding single interface functionality should be our primary goal, any discussions of implementation (aside from feasibility maybe) if we have time.
Some people attending need to leave in the afternoon so be good to get a decent chunk of this done before lunch. If you have additional points we should cover or have alternative ideas please post them here for easy reference.

**Reply by julialawrence** [reply]
Hiya, based on this thread and various conversations in various forums, propose the following agenda;

* User flow for data owner (including integration of the ingestion pipeline)
* User flow for DP user (auth, search, request access, tools)
* Permission mapping between AAD, AWS, Tooling and Catalogue. 

In the language of the doc, this will be your data and your tools.

**Comment by jacobwoffenden** [comment]
I'm closing this discussion as this session has concluded. Actions have been raised.
